### PR TITLE
Remnant hostile hails

### PR DESCRIPTION
**Pull request by temtemy**
This PR will add hostile hails for the Remnant, and varies depending on the interactions of the player (uncontacted, ran away from the blood test, was friends to the Remnant). I did my best to make it match with their attributes (they are isolated human scavengers who escaped the Alpha Wars), but my hails might not be good enough. Feel free to improve them if you want.

(ignore the first two commits, I forgot to separate one commit to another branch, contaminating my master branch)

### CODE REVIEW

File: data/remnant.txt [hunk 1]
```diff
@@ -168,3 +168,115 @@ outfitter "Remnant"
...
+	word
+		"Please "
+		"You must "
+	word
+		"surrender "
+		"desist "
+		"desist and surrender "
+		"surrender yourself "
+		"desist and surrender yourself "
+	word
```
Comment (CyberShadow) [inline data/remnant.txt]: comma?

Comment (temtemy) [reply data/remnant.txt]: Oh, fixing that

**Comment by KiLEdEnNis** [comment]
generally nice, but I think it wouldn't look that good unless there were some more hails (as in, seeing the same combinations every 3 hails isn't good)
plus, usually hails are like

word
"Start"
"Begin"
word
" "
word
"Example"
"Sample"
word (can have a space after them if another sentence will follow after)
"."
"!"

while yours have a space usually at the start or end of a piece of a sentence
try to split such spaces into their own words, to make adding hails easier, which matches most current hails

**Comment by temtemy** [comment]
@KiLEdEnNis I changed my hails to make it easier to add more. I also added another hail and some combinations. Feel free to make more suggestions to my PR. :) (I am really new in hails stuff, I wish there was a guide in the wiki for that)

File: data/remnant.txt [hunk 2]
```diff
@@ -211,12 +226,33 @@ phrase "uncontacted remnant hostile hail"
 	word
 		". "
 	word
-		"You have no chance against us, so leave us alone."
+		"You "
+	word
+		"have no chance"
+		"have little chance"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: should be a word " " before this word

File: data/remnant.txt [hunk 3]
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
 		"Please do not "
 		"Do not "
 	word
-		"be afraid."
+		"be afraid"
+	word
+		" "
+	word
+		"of us"
+		"of the Remnant"
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: should be another word after this to add an interpunction (word ".")

File: data/remnant.txt [hunk 4]
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
...
+	word
+		"of us"
+		"of the Remnant"	
+
+phrase "angered remnant hostile hail"
+	word
+		"We"
+		"The Remnant"
+		"Our ship"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: should be a word " " before this one

File: data/remnant.txt [hunk 5]
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
...
+		"following"
+		"intercepting"
+		"attacking"
+	word
+		" "
+	word
+		"you"
+		"your ship"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: another word " " before this 1

File: data/remnant.txt [hunk 6]
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
...
+		"if you will"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: word " "

File: data/remnant.txt [hunk 7]
```diff
@@ -234,11 +270,60 @@ phrase "angered remnant hail"
...
 
 phrase "contacted remnant hostile hail"
 	word
-		"You human are just like the "
+		"You "
+	word
+		"human"
+		"pure one"
+		"unaltered"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: word " "

File: data/remnant.txt [hunk 8]
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
...
-		"You must be "
+		"You"
 	word
-		"destroyed."
-		"killed."
+		" "
+	word
+		"should be"
+		"must be"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: word " "

File: data/remnant.txt [hunk 9]
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
...
+		"destroyed"
+		"killed"
+		"out of our space"
+		"gone"
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: missing interpunction at the end of this hail
add a word "." (and/or "!")

File: data/remnant.txt [hunk 10]
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
...
 
 phrase "contacted remnant hostile disabled hail"
 	word
-		"Human, if you "
+		"Human,"
+		"Brother,"
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: the game never assumes the players' gender, so this one shouldn't exist as it assumes a male

File: data/remnant.txt [hunk 11]
```diff
@@ -248,14 +333,25 @@ phrase "contacted remnant hostile hail"
...
+		"Pure one,"
+	word
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: word " "

File: data/remnant.txt [hunk 12]
```diff
@@ -274,11 +370,17 @@ phrase "contacted remnant hostile disabled hail"
...
+		"Please"
+	word
+		" "
+	word
+		"have mercy to me,"
+	word
+		" "
 	word
 		"human"
 		"brother"
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: also take this out to keep the game gender-neutral

File: data/remnant.txt [hunk 13]
```diff
@@ -168,3 +168,221 @@ outfitter "Remnant"
...
+	word
+		"destroyed"
+		"killed"
+		"out of our space"
+		"gone"
+
+phrase "contacted remnant hostile disabled hail"
+	word
+		"Human"
+		"Brother"
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: initial comment was hidden in a "show outdated", even though it's not, so I'll repeat:
the game never assumes the player gender, while this assumes male, take this word out

File: data/remnant.txt [hunk 14]
```diff
@@ -168,3 +168,221 @@ outfitter "Remnant"
...
+		"Please"
+	word
+		" "
+	word
+		"have mercy to me,"
+	word
+		" "
+	word
+		"human"
+		"brother"
```
Comment (KiLEdEnNis) [inline data/remnant.txt]: > the game never assumes the player gender, while this assumes male, take this word out

same here

**Comment by KiLEdEnNis** [comment]
some feedback is hidden in "show outdated" things for no reason, check those as well

File: data/remnant.txt [hunk 15]
```diff
@@ -274,11 +370,17 @@ phrase "contacted remnant hostile disabled hail"
 		
 phrase "contacted remnant hostile disabled hail"
 	word
-		"Please have mercy to me, "
+		"Please"
+	word
+		" "
+	word
+		"have mercy to me,"
```
[the remark this answers is not in the document]
Comment (temtemy) [reply data/remnant.txt]: A strict search on DuckDuckGo (searched for "have mercy to me", with quotation marks) reveals no results. So yes, this should be "have mercy on me".

**Comment by Amazinite** [comment]
I personally don't see the need for different types of hostile hails for every different possible scenario with the Remnant thus far. For example, the "angered" hails for the scenario where the player refuses to give blood suggest that the player can redo their choice, which they can't.

**Comment by KiLEdEnNis** [comment]
> suggest that the player can redo their choice, which they can't.

blocking the full Remnant stuff based on 1 choice is a bit bad to me though, we could create some "second" chance (where it's explicitly stated that if the player does it again they'll be locked out)
it's not for this PR but I don't see why we shouldn't add it sometime

**Comment by Amazinite** [comment]
That's why I mentioned 
>every different possible scenario with the Remnant *thus far*.

The Remnant are very bare bones in their story so far, so even aside from the unneeded complexity of these hails, things could change that make them obsolete.

File: data/remnant.txt [hunk 16]
```diff
@@ -168,3 +168,243 @@ outfitter "Remnant"
...
+		"have no chance"
+		"have little chance"
+	word
+		" "
+	word
+		"against us"
+		"on our worlds"
+		"on our planets"
+		"on Ember Waste"
+		"on this region of the galaxy"
```
Comment (endless-sky) [inline data/remnant.txt]: That should be "in the Ember Waste" and "in this region..."

File: data/remnant.txt [hunk 17]
```diff
@@ -168,3 +168,243 @@ outfitter "Remnant"
...
+phrase "angered remnant hostile hail"
+	word
+		"We"
+		"The Remnant"
+		"Our ship"
+	word
+		" "
+	word
+		"will"
+		"is going to"
```
Comment (endless-sky) [inline data/remnant.txt]: "We is going to..." doesn't work grammatically.

**Comment by temtemy** [comment]
This comment seems unrelated to this PR, but can I ask how to rebase my fork's master with this repo's master? To be clear, make it even with this master (no ahead, no behind). Otherwise, I have to delete my fork, which will also delete my remnant branch, and close this PR. Or I can wait until this PR gets merged/closed, which might take weeks, but I am a bit impatient. :) Or maybe there is another way?

**Comment by MCOfficer** [comment]
@temtemy in a clone of your fork (i chose master for this example):
```
// add the main repo as upstream
$ git remote add upstream https://github.com/endless-sky/endless-sky.git

// update master from upstream
$ git checkout master
$ git pull upstream master

// push back to your fork
$ git push
```

**Comment by temtemy** [comment]
Oh no, this is why I asked, because my fork is ahead of master. I will still try though, it might work.

**Comment by temtemy** [comment]
Okay, atleast in comparisons with my fork and upstream are less weird. Is there a way to delete commits (maybe using git)?

**Comment by MCOfficer** [comment]
locally, you can soft-reset your branch back to before whatever you want to revert, then re-issue all commits.
If all commits after the one to be removed are disosable, simply hardreset your branch and it'll all be gone.
if nothing else helps, create a new branch that is uptodate with upstream and cherrypick the commits you want to keep - that's a worst-case solution, though.

**Comment by temtemy** [comment]
My master branch don't have commits to be kept, so I hard-reset to the state before I committed and forced update the remote repo. It is now even with master :)

**Comment by Amazinite** [comment]
Seeing as how #3761 is on the verge of being merged and would conflict with this (as mentioned previously, the angered hails suggest a second chance, but even with #3761's second chance, the only time you would actually see these angered hails is if you already declined that second chance), I'm closing this. I'd suggest speaking with @Zitchas about implementing these in his next PR relating to the Remnant.

**Pull request closed without merging**